src/tests/test_upgrade_agent.py

```diff
@@ -0,0 +1,201 @@
+import datetime
+import json
+import os
+import re
+import subprocess
+from typing import List
+
+import kubernetes.client
+import kubernetes.config
+import kubernetes.dynamic
+import pytest
+from junit_report import JunitTestSuite
+
+from service_client import log
+from tests.base_test import BaseTest
+
+
+class TestUpgradeAgent(BaseTest):
+    @pytest.fixture(scope="session")
+    def namespace(self) -> str:
+        return os.environ.get("NAMESPACE", "assisted-isntaller")
+
+    @pytest.fixture(scope="session")
+    def k8s_client(self) -> kubernetes.dynamic.DynamicClient:
+        return kubernetes.dynamic.DynamicClient(
+            kubernetes.client.api_client.ApiClient(
+                configuration=kubernetes.config.load_kube_config(),
+            ),
+        )
+
+    def _skopeo(self, args: List[str]) -> str:
+        """
+        Runs the 'skopeo' command with the given command line arguments. Raises an exception if the command finishes
+        with a non zero exit code. Returns a string containing whatever the command wrote to the standard output stream.
+        """
+        return subprocess.check_output(["skopeo"] + args).decode()
+
+    def _get_image_repo(self, image: str) -> str:
+        """
+        Calculates the image repository from the given image reference. For example, if the image is
+        'quay.io/my/image:v1.23' it will return 'quay.io/my/image'.
+        """
+        return re.sub(r":.*$", "", image)
```

Comment:
The images reference we have in the CI use digests (instead of tags),  the regex should still work (there's a colon before the digest sha) just note that you will probably get something like this:
registry.build02.ci.openshift.org/ci-op-w23ww3fj/pipeline@sha256
Unsure there's any value in it

Reply:
Replaced with `r"[@:].*$"`, that should address this issue.